bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/installcomponents/handler/InstallCommandHandler.java

```diff
@@ -0,0 +1,145 @@
+	{
+		this.name = name;
+		this.namespace = namespace;
+		this.version = version;
+	}
+
+	public void executeInstallCommand() throws Exception
+	{
+		activateIDFConsoleView();
+		if (!NewProjectHandlerUtil.installToolsCheck())
+		{
+			return;
+		}
+		Map<String, String> envMap = new IDFEnvironmentVariables().getEnvMap();
+		IProject selectedProject = EclipseUtil.getSelectedProjectInExplorer();
+
+		Path pathToProject = new Path(selectedProject.getLocation().toString());
+		List<String> commands = new ArrayList<>();
+		commands.add(IDFUtil.getIDFPythonEnvPath());
+		commands.add(IDFUtil.getIDFPythonScriptFile().getAbsolutePath());
+		commands.add("add-dependency"); //$NON-NLS-1$
+		commands.add(namespace.concat("/").concat(name.concat("==").concat(version))); //$NON-NLS-1$ //$NON-NLS-2$
+
+		console.print((runCommand(commands, pathToProject, envMap)));
+
+		selectedProject.refreshLocal(IResource.DEPTH_INFINITE, null);
+	}
+
+	private String runCommand(List<String> arguments, Path workDir, Map<String, String> env)
+	{
+		String exportCmdOp = ""; //$NON-NLS-1$
+		ProcessBuilderFactory processRunner = new ProcessBuilderFactory();
+		try
+		{
+			IStatus status = processRunner.runInBackground(arguments, workDir, env);
+			if (status == null)
+			{
+				Logger.log(IDFCorePlugin.getPlugin(), IDFCorePlugin.errorStatus("Status can't be null", null)); //$NON-NLS-1$
+				return IDFCorePlugin.errorStatus("Status can't be null", null).toString(); //$NON-NLS-1$
+			}
+
+			// process export command output
+			exportCmdOp = status.getMessage();
+			Logger.log(exportCmdOp);
+		}
+		catch (Exception e1)
+		{
+			Logger.log(IDFCorePlugin.getPlugin(), e1);
+		}
+		return exportCmdOp;
+	}
+
+	private void activateIDFConsoleView()
+	{
+		// Create Tools console
+		MessageConsole msgConsole = findConsole(Messages.IDFToolsHandler_ToolsManagerConsole);
+		msgConsole.clearConsole();
+		console = msgConsole.newMessageStream();
+		msgConsole.activate();
+		// Open console view so that users can see the output
+		openConsoleView();
+	}
+
+	/**
+	 * Find a console for a given name. If not found, it will create a new one and return
+	 * 
+	 * @param name
+	 * @return
+	 */
+	private MessageConsole findConsole(String name)
+	{
+		ConsolePlugin plugin = ConsolePlugin.getDefault();
+		IConsoleManager conMan = plugin.getConsoleManager();
+		IConsole[] existing = conMan.getConsoles();
+		for (int i = 0; i < existing.length; i++)
+		{
+			if (name.equals(existing[i].getName()))
+				return (MessageConsole) existing[i];
+		}
+		// no console found, so create a new one
+		MessageConsole myConsole = new MessageConsole(name, null);
+		conMan.addConsoles(new IConsole[] { myConsole });
+		return myConsole;
+	}
+
+	private void openConsoleView()
+	{
+		try
+		{
+			PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage()
+					.showView(IConsoleConstants.ID_CONSOLE_VIEW);
+		}
+		catch (PartInitException e)
+		{
+			Logger.log(e);
+		}
+	}
+
+}
```

Comment:
Can we use `new IDFConsole().getConsoleStream()` to avoid duplicate code?

Reply:
Done